src/server/editorServices.ts

```diff
@@ -1743,7 +1743,9 @@ namespace ts.server {
 
             return project?.isSolution() ?
                 project.getDefaultChildProjectFromSolution(info) :
-                project;
+                project && projectContainsInfoDirectly(project, info) ?

```

Comment:
Do the other callers of `getConfigFileNameForFile` and `findConfiguredProjectByProjectName` need this check too?

Reply:
No.. this is the change to determine default project for the file...

Reply:
I'm not sure I follow.  Why is direct containment more important for the default project than for other projects?

Reply:
findConfiguredProjectByProjectName  project finds the configured project by the given name and has nothing to do with scriptinfo. 
Similary getConfigFileNameForFile  finds config file name by traversing directories for tsconfig/jsconfig json files. Nothing else matters.

This function is what determines the default project for given script info and hence the additional logic